### PR TITLE
[SMALLFIX] Fix bash style in deploy/vagrant

### PR DESCRIPTION
**Pull request by ifcharming**
This PR fixes the following shell styles under deploy/vagrant:

1) Add missing bash headers.
2) Quote variables; prefer "${var}" over "$var".

### CODE REVIEW

File: deploy/vagrant/bin/run_docker.sh [hunk 1]
```diff
@@ -21,8 +21,8 @@ cat files/id_rsa.pub |awk '{print $1, $2, "Generated by vagrant"}' > files/autho
 vagrant up --provider=docker --no-provision
 
 HOSTS=`vagrant ssh-config 2>/dev/null |grep -w Host |awk '{print $2}'`
-for h in $HOSTS; do
- addr=`vagrant ssh $h -c "ifconfig eth0" |grep -w inet|awk '{print $2}'`
+for h in ${HOSTS}; do
```
Comment (jsimsa) [inline deploy/vagrant/bin/run_docker.sh]: let's call the variable host

File: deploy/vagrant/bin/run_openstack.sh [hunk 2]
```diff
@@ -17,8 +17,8 @@ if [ ! -f files/id_rsa ]; then
 fi
 
 HOSTS=`vagrant ssh-config 2>/dev/null |grep -w Host |awk '{print $2}'`
-for h in $HOSTS; do
- addr=`vagrant ssh $h -c "ifconfig eth0" |grep -w inet|awk '{print $2}'`
+for h in ${HOSTS}; do
```
Comment (jsimsa) [inline deploy/vagrant/bin/run_openstack.sh]: ditto

File: deploy/vagrant/provision/roles/lib/files/device.sh [hunk 3]
```diff
@@ -20,29 +20,29 @@
 
 # Ansible will mount ephemeral disk0 to /media/ephemeral# by default, we would like to mount to /disk#
 ephemeral=`ls /media | egrep '^ephemeral'`
-for d in $ephemeral; do sudo umount /media/$d; done
+for d in ${ephemeral}; do sudo umount /media/${d}; done
```
Comment (jsimsa) [inline deploy/vagrant/provision/roles/lib/files/device.sh]: replace d with device or disk

File: deploy/vagrant/provision/roles/lib/files/device.sh [hunk 4]
```diff
@@ -20,29 +20,29 @@
...
 
 # Solution to determine the devices that haven't been mounted is:
 # find those in the set of the user specified devices but not in the results of `df` yet
 dev=`ls /dev | egrep '^sd|^hd|^xvd' | grep -v da`
 # Ansible will create symbolic links for /dev/xvd? to /dev/sd?, resolve the links
-real_dev=`for d in $dev; do readlink -f /dev/$d; done`
-dev=`echo $real_dev | sed "s/\/dev\///g" | tr ' ' '\n' | sort | uniq`
+real_dev=`for d in ${dev}; do readlink -f /dev/${d}; done`
```
Comment (jsimsa) [inline deploy/vagrant/provision/roles/lib/files/device.sh]: ditto

File: deploy/vagrant/provision/roles/ufs_hadoop1/files/config.sh [hunk 5]
```diff
@@ -9,22 +9,22 @@ NODES=`cat /vagrant/files/workers`
 # setup hadoop
 rm -f /hadoop/conf/slaves
 for i in ${NODES[@]}; do
- echo $i >> /hadoop/conf/slaves
+ echo ${i} >> /hadoop/conf/slaves
```
Comment (jsimsa) [inline deploy/vagrant/provision/roles/ufs_hadoop1/files/config.sh]: fix indentation

File: deploy/vagrant/provision/roles/ufs_hadoop1/files/config.sh [hunk 6]
```diff
@@ -9,22 +9,22 @@ NODES=`cat /vagrant/files/workers`
 # setup hadoop
 rm -f /hadoop/conf/slaves
 for i in ${NODES[@]}; do
```
Comment (jsimsa) [inline deploy/vagrant/provision/roles/ufs_hadoop1/files/config.sh]: s/i/node

File: deploy/vagrant/provision/roles/ufs_hadoop2/files/config.sh [hunk 7]
```diff
@@ -9,11 +9,11 @@ NODES=`cat /vagrant/files/workers`
 # setup hadoop
 rm -f /hadoop/etc/hadoop/slaves
 for i in ${NODES[@]}; do
```
Comment (jsimsa) [inline deploy/vagrant/provision/roles/ufs_hadoop2/files/config.sh]: s/i/node

**Comment by jsimsa** [comment]
@ifcharming awesome! left some comments ... in general it would be great if we could apply more of https://google.github.io/styleguide/shell.xml

File: deploy/vagrant/bin/run_docker.sh (hunk 1 above)
Comment (ifcharming) [inline deploy/vagrant/bin/run_docker.sh]: Done.

File: deploy/vagrant/bin/run_openstack.sh (hunk 2 above)
Comment (ifcharming) [inline deploy/vagrant/bin/run_openstack.sh]: Done.

File: deploy/vagrant/provision/roles/lib/files/device.sh (hunk 3 above)
Comment (ifcharming) [inline deploy/vagrant/provision/roles/lib/files/device.sh]: Done.

File: deploy/vagrant/provision/roles/lib/files/device.sh (hunk 4 above)
Comment (ifcharming) [inline deploy/vagrant/provision/roles/lib/files/device.sh]: Done.

File: deploy/vagrant/provision/roles/ufs_hadoop1/files/config.sh (hunk 6 above)
Comment (ifcharming) [inline deploy/vagrant/provision/roles/ufs_hadoop1/files/config.sh]: Done.

File: deploy/vagrant/provision/roles/ufs_hadoop2/files/config.sh (hunk 7 above)
Comment (ifcharming) [inline deploy/vagrant/provision/roles/ufs_hadoop2/files/config.sh]: Done.

**Comment by ifcharming** [comment]
Thanks @jsimsa for the review.
Agreed, and I'll also investigate if it's easy to apply checkstyle for bash scripts.

File: deploy/vagrant/provision/roles/lib/files/device.sh [hunk 8]
```diff
@@ -20,29 +20,29 @@
...
+to_mount=`printf "%s\n" ${mounted} ${mounted} ${dev} | sort | uniq -u`
 echo "devices known by kernel but not mounted yet: "
-printf "%s\n" $to_mount
+printf "%s\n" ${to_mount}
 
 # format disk, sequentially mount to /disk0, /disk1, ...
 n=0
-for disk in $to_mount; do
+for disk in ${to_mount}; do
  sudo mkfs.ext4 -F "/dev/$disk" # format to ext4 even if it has been formated before
```
Comment (uronce-cc) [inline deploy/vagrant/provision/roles/lib/files/device.sh]: The following `disk` is also not quoted in `${}`.

File: deploy/vagrant/provision/roles/lib/files/maven.sh [hunk 9]
```diff
@@ -13,14 +13,14 @@ cd /vagrant/shared
 IFS='-' read -a array <<< "$MAVEN_FN"
 MAVEN_DIR=/vagrant/shared/${array[0]}-${array[1]}-${array[2]}
 
-if [ -d $MAVEN_DIR ] && [ `readlink -f /usr/bin/mvn` == "$MAVEN_DIR/bin/mvn" ]; then 
+if [ -d ${MAVEN_DIR} ] && [ `readlink -f /usr/bin/mvn` == "$MAVEN_DIR/bin/mvn" ]; then
```
Comment (uronce-cc) [inline deploy/vagrant/provision/roles/lib/files/maven.sh]: `MAVEN_DIR` in the second `[]` is not quoted by `{}`

**Comment by uronce-cc** [comment]
@ifcharming Thanks for cleaning up the style! There are two places left to be restyled. Otherwise, LGTM.

File: deploy/vagrant/provision/roles/lib/files/device.sh (hunk 8 above)
Comment (ifcharming) [inline deploy/vagrant/provision/roles/lib/files/device.sh]: Done.

File: deploy/vagrant/provision/roles/lib/files/maven.sh (hunk 9 above)
Comment (ifcharming) [inline deploy/vagrant/provision/roles/lib/files/maven.sh]: Done.

**Comment by ifcharming** [comment]
Thanks @uronce-cc , comments addressed.
PTAL

**Comment by uronce-cc** [comment]
LGTM

File: deploy/vagrant/provision/roles/lib/files/device.sh [hunk 10]
```diff
@@ -20,32 +20,32 @@
...
 
 # format disk, sequentially mount to /disk0, /disk1, ...
 n=0
-for disk in $to_mount; do
- sudo mkfs.ext4 -F "/dev/$disk" # format to ext4 even if it has been formated before
+for disk in ${to_mount}; do
+ sudo mkfs.ext4 -F "/dev/${disk}" # format to ext4 even if it has been formated before
  sudo mkdir -p "/disk$n"
- sudo mount "/dev/$disk" "/disk$n"
+ sudo mount "/dev/${disk}" "/disk$n"
```
Comment (apc999) [inline deploy/vagrant/provision/roles/lib/files/device.sh]: `${n}` too?
and the line below

File: deploy/vagrant/util/aws_parallel_perf_test.sh [hunk 11]
```diff
@@ -11,14 +11,14 @@ else
...
  # differently from sed on linux, this is a walk around
  real_init_yml=`readlink init.yml`
- $SED "/ - /d" $real_init_yml
- $SED "s/^Total:.*/Total: $n/g" $real_init_yml
- ln -fs $real_init_yml init.yml
- for i in `seq 1 $n`; do
+ ${SED} "/ - /d" ${real_init_yml}
+ ${SED} "s/^Total:.*/Total: $n/g" ${real_init_yml}
+ ln -fs ${real_init_yml} init.yml
+ for i in `seq 1 ${n}`; do
```
Comment (apc999) [inline deploy/vagrant/util/aws_parallel_perf_test.sh]: `${i}` below?

**Comment by apc999** [comment]
left two minor comments

File: deploy/vagrant/provision/roles/lib/files/device.sh (hunk 10 above)
Comment (ifcharming) [inline deploy/vagrant/provision/roles/lib/files/device.sh]: Done.

File: deploy/vagrant/util/aws_parallel_perf_test.sh (hunk 11 above)
Comment (ifcharming) [inline deploy/vagrant/util/aws_parallel_perf_test.sh]: Done.

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**